templates/Angular2Spa/ClientApp/app/app.module.client.ts

```diff
@@ -0,0 +1,21 @@
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { FormsModule } from '@angular/forms';
+import { HttpModule } from '@angular/http';
+import { sharedConfig } from './app.module';
```

Comment:
BTW I tried to do this, but hit a roadblock. If the shared `NgModule` includes the routing config (which of course you would want it to), then it has to be able to compile all the templates for all the components. But to be able to compile them, it needs to reference either `BrowserModule` or `ServerModule` (otherwise it fails with errors like `ngIf is unknown`). But it can't reference either of those, because it needs to work with both server and client.

So is there any reasonable way to do this? Please let me know if there is. It doesn't seem like and `NgModule` can possibly be portable across both server and client, which is an unfortunate limitation.